backend/song_api/views.py

```diff
@@ -27,14 +29,72 @@ def trackDetails(request) -> Response:
 
     """
-    if models.Song.objects.filter(songName=request.data['search']).exists():
-        song = models.Song.objects.get(songName=request.data['search'])
+    all_songs = models.Song.objects.all()
+    song_names = [song.songName for song in all_songs]
+    search_query = request.data['search']
+    matched_song, score = process.extractOne(search_query, song_names, scorer=fuzz.token_set_ratio)
+
+    if score>=80:
+        song = models.Song.objects.get(songName=matched_song)
         if (timezone.now() - song.dateCreated).days <= 7:
             serializer = serializers.SongSerializer(song)
             recommended_songs = song.recommendedSongs.all().order_by('-popularity')
             recommendation = [s.songName for s in recommended_songs]
             artistNetwork = song.artistName.all()
-            return Response({"Recommended Songs": recommendation, "Artist Network": artistNetwork})
+            networkedArtistData = [{"artistName": a.artistName, "artistId": a.artistId} for a in artistNetwork]
+            networkedArtistName = [a['artistName'] for a in networkedArtistData][:11]
+            networkedArtistID = [a['artistId'] for a in networkedArtistData][:11]
+            return Response({"Recommended Songs": recommendation, "Song Name": song.songName, "Networked Artist": networkedArtistName[:11], "Networked Artist ID": networkedArtistID})
+        else:
+            spotifyInstance = spotifyAPI.SpotifyAPI(request.data)
+            songId, songName, collaboratedArtists, topTracks = spotifyInstance.getSongDetails()
+
+            if topTracks == "Error: No song found with the given name.":
+                return Response({"Error": topTracks})
+
+            graph = egoNetwork.ConstructGraph(collaboratedArtists)
+            artistEgoNEtwork = graph.extractArtists()
+            songPopularity = topTracksFromEgoNetwork(artistEgoNEtwork, topTracks)
+
+            networkedArtistID = list(collaboratedArtists.keys())
+            
+            recommendedSongsName = [data['name'] for data in songPopularity][:10]
+            recommendedSongsId = [data['id'] for data in songPopularity][:10]
+            recommendedSongsPopularity = [data['popularity'] for data in songPopularity][:10]
+
+            networkedArtistName = spotifyInstance.networkedArtist(networkedArtistID)
+
+            song.songId = songId
+            song.songName = songName
+            song.dateCreated = timezone.now()
+            song.save()
+
+            song.recommendedSongs.clear()
+            song.artistName.clear()
+
+            for id, name, songPopularity in zip(recommendedSongsId, recommendedSongsName, recommendedSongsPopularity):
+                if models.RecommendedSong.objects.filter(songId=id).exists(): 
+                    suggestion = models.RecommendedSong.objects.get(songId=id)
+                else:
+                    suggestion = models.RecommendedSong.objects.create(songId=id, songName=name, popularity=songPopularity)
```

Comment:
Please, try to make functions smaller and break them into more discrete modules that follow the functionality steps.

Reply:
Carried out the required changes. Please review it once.